### PR TITLE
Friendlyname

### PR DESCRIPTION
**Pull request by BKristenssonAlfsson**
Friendlyname showing in Navbar and dropdown in Settings

### CODE REVIEW

**Review by ivelin — CHANGES_REQUESTED** [review]
Good start, but UX needs some work. See screenshots below. It should be immediately clear to the user what the added value is of this PR.

- Navbar text goes past the navbar boundaries. It sometimes shows "connected to :" (blank, nothing). See images with different display form factors - responsive, mobile.
- As we are moving towards multiple edge device management, we need to reconsider the purpose of this navbar info. What do we want it to show when there are more than one connected devices?
- The settings screen now looks confusing. See screenshots. Remote device section is blank and it has a blank dropdown.
- We need unit tests added for this feature to document programmatically what it's intended purpose is.
- This branch needs to be merged with upstream master since its behind now. I did the merge locally without any issues.
- Please follow the commit lint guidelines to avoid the red flag. "feat: display edge friendly name."


Screen Shot 2020-12-07 at 2 23 13 PM
Screen Shot 2020-12-07 at 2 20 59 PM
Screen Shot 2020-12-07 at 2 22 31 PM
Screen Shot 2020-12-07 at 2 23 26 PM
Screen Shot 2020-12-07 at 2 22 47 PM
Screen Shot 2020-12-07 at 2 23 08 PM

**Review by ivelin — CHANGES_REQUESTED** [review]
See a few comments inline.

File: src/store/pnp.js [hunk 1]
```diff
@@ -133,6 +145,20 @@ const mutations = {
   [PEER_FETCH] (state, peerFetch) {
     console.debug('Setting PeerFetch instance.')
     state.peerFetch = peerFetch
+  },
+  [ADD_NEW_FRIENDLY_NAME] (state, edgeFriendlyName) {
```
[the remark this answers is not in the document]
Comment (BKristenssonAlfsson) [reply src/store/pnp.js:149]: Was said that this PR should only have it stored in an array so current session has the details. Subsequent PR would implement Localstorage. This to not generate to big PR.

[the remark this answers is not in the document]
Comment (BKristenssonAlfsson) [reply src/store/pnp.js:149]: Naming are up for debate. as it started with just adding in a friendlyname to a PeerID, I felt it was most appropiate to use that.

File: src/store/pnp.js [hunk 2]
```diff
@@ -133,6 +145,20 @@ const mutations = {
...
+    if (this.peerFriendlyName === undefined) {
+      state.peerFriendlyName.push(edgeFriendlyName)
+    } else {
+      if (!state.peerFriendlyName.some(data => data.edgeFriendlyName === 'My Home Ambianic')) {
```
[the remark this answers is not in the document]
Comment (BKristenssonAlfsson) [reply src/store/pnp.js:153]: This was as a remote Edge has a name, but your own local Edge, this was decided in the poll a few months ago to have a default value for your own Edge.

The condition is that if I don't check if the device is already in the array, you can have same PeerID multiple times inserted in the array / localstorage.

This generated a bug that escalated, and in the end, it had to be resolved somehow. I did try resolve it in pnp.js too, with mediocre results when it actually could push it in even if it was stored.

File: src/store/pnp.js [hunk 3]
```diff
@@ -133,6 +145,20 @@ const mutations = {
...
+        // The Ambianic Peer ID is already in the store, DO NOTHING
+      } else {
+        state.peerFriendlyName.push(edgeFriendlyName)
```
[the remark this answers is not in the document]
Comment (BKristenssonAlfsson) [reply src/store/pnp.js:156]: Localstorage was decided to be a subsequent PR when I started this

File: src/store/pnp.js [hunk 4]
```diff
@@ -149,7 +175,9 @@ async function discoverRemotePeerId ({ peer, state, commit }) {
   if (state.remotePeerId) {
     return state.remotePeerId
   } else {
-  // first try to find the remote peer ID in the same room
+    // Set the user to be default value
+    commit(CURRENT_ACTIVE_USER, 'My Home Ambianic')
```
[the remark this answers is not in the document]
Comment (BKristenssonAlfsson) [reply src/store/pnp.js:179]: In the poll, it was voted to have the current active Edge friendlyname in the navbar. Hence the implementation

**Comment by BKristenssonAlfsson** [comment]
Closed as the scope was drastically changed and became an epic

**Pull request closed without merging**